rfcs/0000-shareable-courses.md

```diff
@@ -0,0 +1,150 @@
+
+# Alternatives
+
+## No formal support for sharing questions
+
+This is the current state of PrairieLearn. "Sharing" a question involves manually copying all of its files from one course to another. From a technical perspective, this solution is simple, as it requires no special treatment in PrairieLearn code. However, it scales poorly as it requires a lot of manual action to propagage question enhancements or fixes.
+
+## Share questions directly from courses
+
+Instead of having a special entity that stores shared questions, one could share questions directly from an existing course. From a technical perspective this is likely easier than the proposal of dedicated shareable courses.
+
+The first drawback to this solution is that it makes it hard to ensure the previously-mentioned consistency guarantees. Since a normal course is a self-contained unit, questions can be renamed and deleted at will as long as all references to the question within the course change at the same time. This is by design, as it allows courses to reorganize their course material as time goes on. However, with dedicated shareable courses, a question's ID becomes a part of the "interface" of the course.
+
+The second drawback is less technical and more organizational. Specifically, it does nothing to discourage tying a question too closely to the needs of a particular course/institution or making changes that may be unexpected or undesired by consumers of the question. While we can't enforce these things at the technical level, the hope is that by having shared questions located in a separate repo, question editors will think more carefully about how they design questions and the changes they make to them over time.
+
+# Potential future enhancements
+
+The following features are not necessary for an MVP version of this feature, but are called out for the sake of discussion.
+
+## Question deprecation
+
+To ensure safety guarantees, once a question in a shareable course exists, it cannot be renamed or deleted. However, as shareable courses eveolve over the course of many years, certain questions may become outdated or broken and there may not be sufficient resources to update them. To make it easy for this to be communicated to consumers of questions, a "deprecated" flag could be added to a question's `info.json` file:
+
+```json
+{
+  "uuid": "754dac56-fc1f-4888-a74a-1b365fd48376",
+  "deprecated": true
+}
+```
+
+Deprecated questions could render a warning in instructor views of the question, on a question's row in an assessment overview, or in other useful locations across PrairieLearn.
+
+# Unresolved questions and concerns
+
+## Feature name
+
+With the existence of assessment sets, "set" might become too much of an overloaded term if used for this feature. A name like "question collections" may be more appropriate.
+
```

Comment:
I think this subsection can be deleted, now that we've changed to calling this a "shareable" course.

```suggestion

```